src/views/Migrate.vue

```diff
@@ -19,28 +41,45 @@ import { Component, Vue } from 'vue-property-decorator';
 import { AccountInfo } from '@/lib/AccountInfo';
 import { WalletStore } from '@/lib/WalletStore';
 import { WalletInfo, WalletType } from '@/lib/WalletInfo';
-import { SmallPage } from '@nimiq/vue-components';
+import { SmallPage, PageHeader, PageBody, PageFooter, AccountList } from '@nimiq/vue-components';
 import Network from '@/components/Network.vue';
 import Loader from '@/components/Loader.vue';
-import { KeyguardClient } from '@nimiq/keyguard-client';
+import KeyguardClient from '@nimiq/keyguard-client';
 import { ACCOUNT_DEFAULT_LABEL_LEGACY } from '@/lib/Constants';
 import { ContractInfo } from '@/lib/ContractInfo';
+import { Static } from '@/lib/StaticStore';
+import { SimpleRequest } from '@/lib/PublicRequestTypes';
 
-@Component({components: {SmallPage, Loader, Network}})
+@Component({components: {SmallPage, PageHeader, PageBody, PageFooter, AccountList, Loader, Network}})
 export default class Migrate extends Vue {
+    @Static private request!: SimpleRequest;
+
     private title: string = 'Migrating your accounts';
     private status: string = 'Connecting to Keyguard...';
     private state: Loader.State = Loader.State.LOADING;
     private message: string = '';
 
-    private keyguardClient?: KeyguardClient;
+    private legacyAccounts: AccountInfo[] = [];
+    private isMigrating: boolean = false;
+
+    public async created() {
+        const legacyKeys = await this.$rpc.keyguardClient.listLegacyAccounts();
+        this.legacyAccounts = legacyKeys.map((key) => this.legacyKeyInfoObject2AccountInfo(key));
+    }
+
+    private accountSelected(walletId: string, address: string) {
+        // Start export request for this address
+        const request: KeyguardClient.ExportRequest = {
+            appName: this.request.appName,
+            keyId: address,
+            keyLabel: '',
+        };
 
-    public mounted() {
-        this.keyguardClient = this.$rpc.createKeyguardClient();
-        this.run();
+        const client = this.$rpc.createKeyguardClient();
+        client.export(request);
     }
 
-    private async run() {
+    private async runMigration() {
```

Comment:
Why are two methods needed for this? They could be combined into one I think.

Reply:
Do you mean `runMigration` and `doMigration`? Because I liked it better to have a small `try {} catch` loop around a single function call, as opposed to around a whole function block.

Reply:
Yes, I thought as much.